### PR TITLE
publishing updates to deployer including transfers

### PR DESCRIPTION
**Pull request by poiresel**
## Description

Updated deployer to incorporate the new contracts of `transferRegistry` + `preSwapChecker` 

Quick description:
- [X] New features

## Tests
did a smoke test
tested with running migration for development where I put in a customer fungibletoken and weth9 and deployment succeeded as well as checks to the transferRegistry had the write handlers on them.

## Test Coverage
N/A

### CODE REVIEW

File: utils/deployer/migrations/1_initial_migration.js [hunk 1]
```diff
@@ -1,26 +1,54 @@
...
   await Swap.link("Types", Types.address)
-  await deployer.deploy(Swap)
+  await deployer.deploy(Swap, TransferHandlerRegistry.address)
   await deployer.deploy(Indexer, STAKING_TOKEN_ADDRESS)
   await deployer.deploy(DelegateFactory, Swap.address, Indexer.address, '0x0001')
   await deployer.deploy(Wrapper, Swap.address, WETH_ADDRESS)
 
+  await deployer.deploy(PreSwapChecker, WETH_ADDRESS)
   //Deploy Contract-Created Contracts that also need to be verified
   await deployer.deploy(Index)
```
Comment (ejwessel) [inline utils/deployer/migrations/1_initial_migration.js]: While we're at it, did we want to fix this wasteful index I made prior?

File: utils/deployer/migrations/1_initial_migration.js [hunk 2]
```diff
@@ -1,26 +1,54 @@
...
 module.exports = async(deployer, network) => {
 
   network = network.toUpperCase()
+
+
   let STAKING_TOKEN_ADDRESS = process.env[network + "_STAKE"]
   let WETH_ADDRESS = process.env[network + "_WETH"]
 
-  //Deploy Base Contracts 
+  const CKITTY_KIND = '0x9a20483d'
```
Comment (ejwessel) [inline utils/deployer/migrations/1_initial_migration.js]: should we stick this into `order-utils` constants?

File: utils/deployer/migrations/1_initial_migration.js (hunk 1 above)
Comment (ejwessel) [reply utils/deployer/migrations/1_initial_migration.js]: We can also do it with delegate.

Comment (ejwessel) [reply utils/deployer/migrations/1_initial_migration.js]: Note: these are for verification purposes

File: source/delegate/migrations/1_initial_migration.js [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
-module.exports = (deployer, network) => {
-  if (network == 'rinkeby' || network == 'mainnet') {
+module.exports = async(deployer, network) => {
     // fill in the addresses below
     const SWAP_ADDRESS = ''
     const INDEXER_ADDRESS = ''
     const OWNER_ADDRESS = ''
     const TRADE_WALLET_ADDRESS = ''
-    deployer.deploy(Delegate, SWAP_ADDRESS, INDEXER_ADDRESS, OWNER_ADDRESS, TRADE_WALLET_ADDRESS)
+    const FACTORY_PROTOCOL = ''
```
Comment (alicevhenshaw) [inline source/delegate/migrations/1_initial_migration.js]: Should we fill this in? I feel like we have a standard factory protocol that we use

File: source/delegate/migrations/1_initial_migration.js [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+
+  if (network == 'mainnet' || network == 'rinkeby') {
+    await deployer.deploy(DelegateFactory, SWAP_ADDRESS, INDEXER_ADDRESS, FACTORY_PROTOCOL)
+    await deployer.deploy(Delegate, SWAP_ADDRESS, INDEXER_ADDRESS, OWNER_ADDRESS, TRADE_WALLET_ADDRESS, FACTORY_PROTOCOL)
```
Comment (alicevhenshaw) [inline source/delegate/migrations/1_initial_migration.js]: Why are we deploying a delegate? Isn't that what the delegate factory is for? I feel like it'd be better to deploy a factory and then create a delegate from that

Comment (alicevhenshaw) [reply source/delegate/migrations/1_initial_migration.js]: Tbf I see that we dont ever really use this file

Comment (ejwessel) [reply source/delegate/migrations/1_initial_migration.js]: As I understood it, it was difficult to verify the delegate from the delegate factory and so deploying a dummy one to verify was the only way to do it.

File: utils/deployer/README.md [hunk 5]
```diff
@@ -39,5 +39,5 @@ MAINNET_STAKE="0x27054b13b1b798b345b591a4d22e6562d47ea75a"
 
 If looking to verify on a network besides `development`, one can append the verification commands for the contracts that wish to be verified
 ```
-yarn migrate --network rinkeby && sleep 40 && yarn verify Types Wrapper DelegateFactory Indexer Swap flatten/Flattened.sol --network rinkeby
+yarn migrate --network mainnet && sleep 40 && yarn verify Types Wrapper DelegateFactory Indexer Swap TransferHandlerRegistry ERC1155TransferHandler ERC20TransferHandler ERC721TransferHandler KittyCoreTransferHandler flatten/Flattened.sol --network mainnet
```
Comment (ejwessel) [inline utils/deployer/README.md]: nice catch on remembering to do this.

File: source/delegate/migrations/1_initial_migration.js (hunk 3 above)
Comment (ejwessel) [reply source/delegate/migrations/1_initial_migration.js]: I believe this is just for dummy verification for the deployed delegate contact. So, it's ok for it to be empty.

**Comment by dmosites** [comment]
Let's update versions with `beta` so that they're published to NPM with the beta tag.

File: utils/pre-swap-checker/package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@airswap/pre-swap-checker",
-  "version": "0.0.3",
+  "version": "0.1.3",
```
Comment (alicevhenshaw) [inline utils/pre-swap-checker/package.json]: need to be marked as beta

File: source/delegate/package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@airswap/delegate",
-  "version": "1.5.8",
+  "version": "2.6.8-beta.1",
```
Comment (alicevhenshaw) [inline source/delegate/package.json]: I believe it should be beta.0 before beta.1 - just as we would have 2.6.0 before 2.6.1

File: utils/pre-swap-checker/package.json (hunk 6 above)
Comment (poiresel) [reply utils/pre-swap-checker/package.json]: even though it's not being deployed to npm?

File: source/delegate/migrations/1_initial_migration.js (hunk 3 above)
Comment (poiresel) [reply source/delegate/migrations/1_initial_migration.js]: Happy to put in '0x0001' as that's what in the deployer repo as well.

File: source/delegate/migrations/1_initial_migration.js (hunk 4 above)
Comment (poiresel) [reply source/delegate/migrations/1_initial_migration.js]: This was custom deployed to allow the reuse of an existing indexer as opposed to redeploying new contracts.

File: utils/pre-swap-checker/package.json (hunk 6 above)
Comment (alicevhenshaw) [reply utils/pre-swap-checker/package.json]: Yeah that is what was concluded after a discussion on slack. We beta them anyway.

File: utils/debugger/package.json [hunk 8]
```diff
@@ -21,9 +21,9 @@
     "mocha": "^6.2.0"
   },
   "dependencies": {
-    "@airswap/order-utils": "0.3.20-beta.1",
-    "@airswap/swap": "5.4.7-beta.1",
+    "@airswap/order-utils": "0.3.20-beta.0",
+    "@airswap/swap": "5.4.7-beta.0",
     "@airswap/tokens": "0.1.4",
-    "ethers": "^5.0.0-beta.159"
+    "ethers": "^5.0.0-beta.059"
```
Comment (alicevhenshaw) [inline utils/debugger/package.json]: Mistake here from your find and replace

**Pull request merged**